addon/src/main/java/org/vaadin/risto/stepper/AbstractStepper.java

```diff
@@ -28,38 +28,34 @@
     private T value;
 
     public AbstractStepper() {
-        registerRpc(new StepperRpc() {
-
-            @Override
-            public void valueChange(String value) {
-                if (isEnabled() && !isReadOnly()) {
-
-                    try {
-                        T parsedValue = parseStringValue(value);
-                        if (isInvalidValuesAllowed() ||
-                                (parsedValue == null && isNullValueAllowed()) ||
-                                (parsedValue != null &&
-                                         isValidForRange(parsedValue))) {
-                            setValue(parsedValue, true);
-                        }
-
-                        /*
-                         * Client side updates the state before sending the
-                         * event so we need to make sure the cached state is
-                         * updated to match the client. If we do not do this, a
-                         * reverting setValue() call in a listener will not
-                         * cause the new state to be sent to the client. This is
-                         * a problem with Vaadin state caching.
-                         *
-                         * See e.g. http://dev.vaadin.com/ticket/12133
-                         */
-                        getUI().getConnectorTracker()
-                                .getDiffState(AbstractStepper.this)
-                                .put("value", value);
-
-                    } catch (StepperValueParseException e) {
-                        handleParseException(e);
+        registerRpc((StepperRpc) value -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "value" which hides the field declared at line 28. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)